pkg/issuer/acme/dns/dns.go

```diff
@@ -140,6 +141,18 @@ func (s *Solver) solverFor(crt *v1alpha1.Certificate, domain string) (solver, er
 		if err != nil {
 			return nil, fmt.Errorf("error instantiating cloudflare challenge solver: %s", err.Error())
 		}
+	case providerConfig.DigitalOcean != nil:
+		apiTokenSecret, err := s.secretLister.Secrets(s.resourceNamespace).Get(providerConfig.DigitalOcean.Token.Name)
+		if err != nil {
+			return nil, fmt.Errorf("error getting digitalocean token: %s", err.Error())
```

Comment:
There's no need to call `.Error` here (I know other parts of this file do too, they should be fixed as well).

The `%s` format string will automatically call `.Error` for anything that implements error (with a higher priority than calling `.String` even, see [playground](https://play.golang.org/p/gw3XZDu7MkJ))